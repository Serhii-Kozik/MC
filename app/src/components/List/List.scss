.results-container {
    max-width: 980px;
    margin: 0 auto;
    padding: 16px;
    border: 1px solid #f4f4f4;
    background: #ffffff;
    min-height: 80vh;
}