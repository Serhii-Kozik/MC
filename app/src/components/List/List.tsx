import React from 'react';
import { Athlet } from '../../actions';
import ListItem from '../ListItem/ListItem';

import FlipMove from 'react-flip-move';
import './List.scss';

export interface Props {
    athlets: Athlet[];
}

class List extends React.Component<Props> {

    render(){
        // sort the athletes by finish time (those who don't finish yet will be on top) 
        const sortedList = this.props.athlets.sort(function(a,b){
            return a.finish - b.finish;
          });
        // prepare place 
        let place: number = 0;
    
        const listItems = sortedList.map((athlet, index) => {
            // assign place to the athlet
            place = athlet.finish !== null ? place + 1 : place;
                // return athlet table line
                return <ListItem
                        key={athlet.start_nr}
                        index={index}
                        place={place ? place : ''}
                        athletNumber={athlet.start_nr}
                        athletFullName={athlet.full_name}
                        athletFinishTime={athlet.finish}
                    />
        });
        // render athletes in a list
        return( 
            <div className='results-container'>
                <ListItem
                    title={true}
                    place='Place'
                    athletNumber='Start Nr'
                    athletFullName='Name'
                    athletFinishTime='Time'
                />
                <FlipMove // animate in and order change
                staggerDurationBy='30'
                duration={500}
                enterAnimation='accordionVertical'
                leaveAnimation='elevator'
                 >
                    {listItems}
                </FlipMove>
            </div>
        );
    }
}

export default List;