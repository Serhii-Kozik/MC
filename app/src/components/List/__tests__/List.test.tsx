import React from 'react';
import List , { Props } from '../List';
import Enzyme ,{ shallow }from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('List component', () => {

    it('should render correctly', () => {
        const props: Props =  {
            athlets: [
              {
                id: 1,
                code: '123',
                start_nr: '123',
                full_name: 'first name',
                entering: 1,
                finish:2
              },
              {
                id: 2,
                code: '124',
                start_nr: '124',
                full_name: 'first name',
                entering: 2,
                finish:1
              },
            ],
          };

        const component = shallow(<List {...props}/>);
        expect(component).toMatchSnapshot();
    });
});