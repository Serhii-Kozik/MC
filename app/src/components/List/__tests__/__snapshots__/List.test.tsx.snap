// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    athlets={
      Array [
        Object {
          "code": "124",
          "entering": 2,
          "finish": 1,
          "full_name": "first name",
          "id": 2,
          "start_nr": "124",
        },
        Object {
          "code": "123",
          "entering": 1,
          "finish": 2,
          "full_name": "first name",
          "id": 1,
          "start_nr": "123",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ListItem
          athletFinishTime="Time"
          athletFullName="Name"
          athletNumber="Start Nr"
          place="Place"
          title={true}
        />,
        <FlipMovePropConverter
          delay={0}
          disableAllAnimations={false}
          duration={500}
          easing="ease-in-out"
          enterAnimation="accordionVertical"
          getPosition={[Function]}
          leaveAnimation="elevator"
          maintainContainerHeight={false}
          staggerDelayBy={0}
          staggerDurationBy="30"
          typeName="div"
          verticalAlignment="top"
        >
          <ListItem
            athletFinishTime={1}
            athletFullName="first name"
            athletNumber="124"
            index={0}
            place={1}
          />
          <ListItem
            athletFinishTime={2}
            athletFullName="first name"
            athletNumber="123"
            index={1}
            place={2}
          />
        </FlipMovePropConverter>,
      ],
      "className": "results-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "athletFinishTime": "Time",
          "athletFullName": "Name",
          "athletNumber": "Start Nr",
          "place": "Place",
          "title": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ListItem
              athletFinishTime={1}
              athletFullName="first name"
              athletNumber="124"
              index={0}
              place={1}
            />,
            <ListItem
              athletFinishTime={2}
              athletFullName="first name"
              athletNumber="123"
              index={1}
              place={2}
            />,
          ],
          "delay": 0,
          "disableAllAnimations": false,
          "duration": 500,
          "easing": "ease-in-out",
          "enterAnimation": "accordionVertical",
          "getPosition": [Function],
          "leaveAnimation": "elevator",
          "maintainContainerHeight": false,
          "staggerDelayBy": 0,
          "staggerDurationBy": "30",
          "typeName": "div",
          "verticalAlignment": "top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "124",
            "nodeType": "class",
            "props": Object {
              "athletFinishTime": 1,
              "athletFullName": "first name",
              "athletNumber": "124",
              "index": 0,
              "place": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "123",
            "nodeType": "class",
            "props": Object {
              "athletFinishTime": 2,
              "athletFullName": "first name",
              "athletNumber": "123",
              "index": 1,
              "place": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ListItem
            athletFinishTime="Time"
            athletFullName="Name"
            athletNumber="Start Nr"
            place="Place"
            title={true}
          />,
          <FlipMovePropConverter
            delay={0}
            disableAllAnimations={false}
            duration={500}
            easing="ease-in-out"
            enterAnimation="accordionVertical"
            getPosition={[Function]}
            leaveAnimation="elevator"
            maintainContainerHeight={false}
            staggerDelayBy={0}
            staggerDurationBy="30"
            typeName="div"
            verticalAlignment="top"
          >
            <ListItem
              athletFinishTime={1}
              athletFullName="first name"
              athletNumber="124"
              index={0}
              place={1}
            />
            <ListItem
              athletFinishTime={2}
              athletFullName="first name"
              athletNumber="123"
              index={1}
              place={2}
            />
          </FlipMovePropConverter>,
        ],
        "className": "results-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "athletFinishTime": "Time",
            "athletFullName": "Name",
            "athletNumber": "Start Nr",
            "place": "Place",
            "title": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ListItem
                athletFinishTime={1}
                athletFullName="first name"
                athletNumber="124"
                index={0}
                place={1}
              />,
              <ListItem
                athletFinishTime={2}
                athletFullName="first name"
                athletNumber="123"
                index={1}
                place={2}
              />,
            ],
            "delay": 0,
            "disableAllAnimations": false,
            "duration": 500,
            "easing": "ease-in-out",
            "enterAnimation": "accordionVertical",
            "getPosition": [Function],
            "leaveAnimation": "elevator",
            "maintainContainerHeight": false,
            "staggerDelayBy": 0,
            "staggerDurationBy": "30",
            "typeName": "div",
            "verticalAlignment": "top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "124",
              "nodeType": "class",
              "props": Object {
                "athletFinishTime": 1,
                "athletFullName": "first name",
                "athletNumber": "124",
                "index": 0,
                "place": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "123",
              "nodeType": "class",
              "props": Object {
                "athletFinishTime": 2,
                "athletFullName": "first name",
                "athletNumber": "123",
                "index": 1,
                "place": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
